sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java

```diff
@@ -1456,6 +1457,111 @@ public void canDeleteVMSSInstance() throws Exception {
         computeManager.virtualMachineScaleSets().deleteInstances(rgName, vmssName, Collections.singleton(vmss.virtualMachines().list().stream().findFirst().get().instanceId()), false);
     }
 
+    @Test
+    public void canCreateFlexibleVMSS() throws Exception {
+        // create vmss with flexible orchestration type
+        VirtualMachineScaleSetInner options = new VirtualMachineScaleSetInner();
+        options.withOrchestrationMode(OrchestrationMode.FLEXIBLE)
+            .withPlatformFaultDomainCount(1)
+            .withLocation(region.name());
+
+        ResourceGroup resourceGroup = this.resourceManager.resourceGroups().define(rgName)
+            .withRegion(region.name())
+            .create();
+
+        Network network =
+            this
+                .networkManager
+                .networks()
+                .define("vmssvnet")
+                .withRegion(region.name())
+                .withExistingResourceGroup(resourceGroup)
+                .withAddressSpace("10.0.0.0/28")
+                .withSubnet("subnet1", "10.0.0.0/28")
+                .create();
+        final String vmssName = generateRandomResourceName("vmss", 10);
+        LoadBalancer publicLoadBalancer = createHttpLoadBalancers(region, resourceGroup, "1", LoadBalancerSkuType.STANDARD, PublicIPSkuType.STANDARD, true);
+
+        // create resource through raw method
+        VirtualMachineScaleSet vmss = this
+            .computeManager
+            .virtualMachineScaleSets()
+            .define(vmssName)
+            .withRegion(region.name())
+            .withExistingResourceGroup(resourceGroup)
+            .withFlexibleOrchestrationMode()
+            .withSku(VirtualMachineScaleSetSkuTypes.STANDARD_A0)
+            .withExistingPrimaryNetworkSubnet(network, "subnet1")
+            .withExistingPrimaryInternetFacingLoadBalancer(publicLoadBalancer)
+            .withoutPrimaryInternalLoadBalancer()
+            .withPopularLinuxImage(KnownLinuxVirtualMachineImage.CENTOS_8_3)
+            .withRootUsername("jvuser")
+            .withSsh(sshPublicKey())
+            .create();
+
+        Assertions.assertNotNull(vmss.innerModel().virtualMachineProfile());
+        Assertions.assertNotNull(vmss.getPrimaryInternetFacingLoadBalancer());
+        Assertions.assertNotNull(vmss.getPrimaryNetwork());
+        Assertions.assertEquals(vmss.orchestrationMode(), OrchestrationMode.FLEXIBLE);
+
```

Comment:
Maybe add a `vmss.update().withTag(...).apply()` to check that `virtualMachineProfile` not changed after this.

Reply:
Great idea, added not null check on `virtualMachineProfile`.